pkg/registries/adapters/adapter.go

```diff
@@ -130,3 +141,25 @@ func imageToSpec(log *logging.Logger, req *http.Request, apbtag string) (*apb.Sp
 
 	return spec, nil
 }
+
+func isCompatibleVersion(specVersion string, minVersion string, maxVersion string) bool {
+	specFloat, err := strconv.ParseFloat(specVersion, 64)
```

Comment:
Create a unit test for this because I'm getting a sense this could be bad with bad input. This implementation won't allow 1.0.alpha or anything alphanumeric. So let's test with different inputs and expected errors etc.